besu/src/main/java/org/hyperledger/besu/cli/subcommands/operator/BackupState.java

```diff
@@ -0,0 +1,120 @@
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+package org.hyperledger.besu.cli.subcommands.operator;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static org.hyperledger.besu.cli.DefaultCommandValues.MANDATORY_LONG_FORMAT_HELP;
+
+import org.hyperledger.besu.BesuInfo;
+import org.hyperledger.besu.controller.BesuController;
+import org.hyperledger.besu.ethereum.api.query.StateBackupService;
+import org.hyperledger.besu.ethereum.api.query.StateBackupService.BackupStatus;
+import org.hyperledger.besu.ethereum.chain.MutableBlockchain;
+import org.hyperledger.besu.ethereum.eth.manager.EthScheduler;
+import org.hyperledger.besu.ethereum.worldstate.WorldStateStorage;
+import org.hyperledger.besu.metrics.noop.NoOpMetricsSystem;
+
+import java.io.File;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.LockSupport;
+
+import picocli.CommandLine.Command;
+import picocli.CommandLine.Option;
+import picocli.CommandLine.ParentCommand;
+
+@Command(
+    name = "backup-state",
+    description = "Backups up the state and accounts of a specified block.",
+    mixinStandardHelpOptions = true)
+public class BackupState implements Runnable {
+
+  @Option(
+      names = "--block",
+      paramLabel = MANDATORY_LONG_FORMAT_HELP,
+      description = "The block to perform the backup at (default: calculated chain head)",
+      arity = "1..1")
+  private final Long block = Long.MAX_VALUE;
+
+  @Option(
+      names = "--backup-path",
+      required = true,
+      paramLabel = MANDATORY_LONG_FORMAT_HELP,
+      description = "The path to store the backup files.",
+      arity = "1..1")
+  private final File backupDir = null;
+
+  @Option(
+      names = {"--compression-enabled"},
+      description = "Enable data compression",
+      arity = "1")
+  private final Boolean compress = true;
+
+  @ParentCommand private OperatorSubCommand parentCommand;
+
+  @Override
+  public void run() {
+    checkArgument(
+        backupDir.exists() || backupDir.mkdirs(), "Backup directory exists or can be created");
+
+    final BesuController besuController = createBesuController();
+    final MutableBlockchain blockchain = besuController.getProtocolContext().getBlockchain();
+    final WorldStateStorage worldStateStorage =
+        besuController.getProtocolContext().getWorldStateArchive().getWorldStateStorage();
+    final EthScheduler scheduler = new EthScheduler(1, 1, 1, 1, new NoOpMetricsSystem());
```

Comment:
Do we need to create a whole `EthScheduler` for this? Can we just have an `ExecutorService`? Then we can do `CompleteableFuture.supplyAsync(computation, executor)`.

Reply:
We need the EthScheduler to futureproof the StateBackupService.  We may do backup segments in parallel or even pipelined and the service should have access to the needed infrastructure.

Reply:
Right now we are a self-contained subsystem, but on the roadmap is doing these backups from a web service on a live node as well as a disconnected node.  Putting int he eth scheduler now ensures the backup is compatibile with the threading model of the node.